datasets/restaurant.py

```diff
@@ -0,0 +1,232 @@
+import random
+from typing import List, Tuple, Iterator, Callable
+
+from datetime import datetime, timedelta
+from dataclasses import dataclass, field
+from dataset_interfaces.interface import DatasetInterface, TestExample, TestAction, SendMessageAction, WaitAction
+
+from utils.openai import ask_llm, LLMContext, make_user_message
+from goodai.helpers.json_helper import sanitize_and_parse_json
+
+
+@dataclass
+class RestaurantExample(TestExample):
+    cost_callback: Callable[[float], None] = None
+    reset_message: str = "Let's not pretend to be at a restaurant anymore. Please also forget everything about it."
+    score: int = 0
+    max_score: int = 1
+    filler_tokens: int = 1000
+    expected_responses: list[str] = field(default_factory=list)
+    script: List[str] = field(default_factory=lambda: [""])  # Required for compatibility, but no actual script.
+
+    def __post_init__(self):
+        super().__post_init__()
+        assert self.cost_callback is not None, "Dynamic examples require a cost callback."
+        assert self.evaluation_fn is None, "Dynamic examples have their own evaluation function."
```

Comment:
These asserts seem to be generic, but the class they are in is specific. Do you want to try and make a `DyanmicTextExample` class?

Reply:
oh yeah, definitely. That's done but not pushed yet.